be/src/storage/tablet_updates.cpp

```diff
@@ -1319,13 +1319,29 @@ void TabletUpdates::_apply_compaction_commit(const EditVersionInfo& version_info
     size_t total_rows = 0;
     vector<std::pair<uint32_t, DelVectorPtr>> delvecs;
     vector<uint32_t> tmp_deletes;
+
+    // Since value stored in info->inputs of CompactInfo is rowset id
+    // we should get the real max rssid here by segment number
+    int64_t max_src_rssid = -1;
+    for (auto rowsetid : info->inputs) {
+        auto itr = _rowsets.find(rowsetid);
+        if (itr == _rowsets.end()) {
+            string msg = Substitute("rowset not found tablet=$0 rowset=$1", _tablet.tablet_id(), rowsetid);
+            DCHECK(false) << msg;
+            LOG(WARNING) << msg;
+            continue;
+        }
+        uint32_t rowset_max_rssid = rowsetid + itr->second->num_segments() - 1;
+        max_src_rssid = std::max(max_src_rssid, (int64_t) rowset_max_rssid);
+    }
+
```

Comment:
only need to get rowset of with max rowset id, do not need to get all rowsets

Reply:
There is a scene I don't know if it will appear. we have rowset id 1 and 2, rowset 1 has 3 segments and rowset 2 has 1 segment. then we may get rssid from index cache with value 4 for rowset 1.

Reply:
no, rssid will not overlap